contracts/Cryptoxo.sol

```diff
@@ -0,0 +1,254 @@
+    // Several public functions/variables added fore easier understanding on what stage the game is, like current contract balance, current state of OXO board, previous player etc etc
+    // Few rudiments left in the code and are commented but not erased, for me to debug contract later and add more stuff to it
+    
+    address public owner;
+    address public playerRateProposer;
+    address public previousMovePlayer;
+    
+    uint public rateProposed;
+    uint public movesCounter;
+    
+    string empty;
+    
+    // 0x32.. => X
+    mapping (address => string) public playersVal;
+    
+    struct Players {
+        address x;
+        address o;
+    }
+    
+    Players players;
+    
+    // this is the main element in this entired code
+    string[3][3] oxoBoard;
+    
+    //event LoopCount(uint _count);
+    //event ShowValPlayers(address _palyerVal);
+    //event Alert(string _one, string _two, string _three);
+    event ForciblyDraw(string);
+    event ReturnToInintialStateComplete(string);
+    event UserWin(address _winner, uint _amountWon, string _compareMethodName);
+    event GameIsDraw(string _drawGameMsg, uint _amountToPayback, string _goesTo, address _playerX, address _PlayerO);
+    event MoveMade(address _currentPlayer, string _playerVal, uint8 _positionColumn, uint8 _positionRow);
+    event MatchContinius(string _matchContinue);
+    event showOxoBoardString(
+                            string _lineNum, 
+                            string _firstPosition, 
+                            string _secondPosition, 
+                            string _thirdPosition
+                            );
+                            
+    // modifier for function to return ETH in case 
+    // playerRateProposer made an offer
+    // but none accepted the chellenge
+    modifier onlyPlayerRateProposer() {
+        require(msg.sender == playerRateProposer);
+        _;
+    }
+    
+    // modifier to use forciblyDraw() function 
+    // in case something goes wrong and one of players disappears
+    // to manually draw the game
+    modifier onlyOwner() {
+        require(msg.sender == owner);
+        _;
+    }
+    
+    // according to rules onwer can't play the game
+    // owner can only be a reffery 
+    // in case one of players disappears
+    // to manually draw the game
+    modifier onlyNotOwner() {
+        require(msg.sender != owner);
+        _;
+    }
+    
+    
+    
+    // init
+    function Cryptoxo(){
+        // adding owner in case a contract needs a suicide
+        owner = msg.sender;
+        //movesCounter = 0;
+        
+        // creating an empty oxo board;
+        oxoBoard[0] = ["-", "-", "-"];
+        oxoBoard[1] = ["-", "-", "-"];
+        oxoBoard[2] = ["-", "-", "-"];
+    }
+    
+    
+    
+    
+    function offerRate() public payable onlyNotOwner returns (uint){
+        require(this.balance == msg.value);
```

Comment:
So, I can't start to play if the contract has no money?

Reply:
initially, contract has no money. when user call this function - contract receives the ETH. this.balance == msg.value - is to avoid from calling this function when game has started (any msg.value will be less that this.balance, as ETH is sent in transaction, and require condition is checked only after ETH is sent)